### ISSUE TITLE
paper-input followed by ha-icon-button prevent tab from moving focus forward

### ISSUE THREAD

**Issue by JeffersonBledsoe** [home-assistant/frontend#7206]
## Checklist

- [x] I have updated to the latest available Home Assistant version.
- [x] I have cleared the cache of my browser.
- [x] I have tried a different browser to see if it is related to my browser.

## The problem
Following on from #4689, I have noticed that a `paper-input` followed by a `ha-icon-button` causes the focus to not be allowed to move forward. A gif is attached below of trying to tab past the `entity_id` input when configuring the default weather forecast card. The first "TAB" press causes the icon button to the side of the input to display the material ripple, however the button is not focused. Instead, focus is returned to the input, causing the user to be trapped.

![Unable to tab past the entity_id field when configuring a card](https://user-images.githubusercontent.com/30210785/94978896-8fce9800-0517-11eb-818f-a42eb894d9f4.gif)

## Expected behavior
The logical tab order is followed (i.e. the next field is focused). The user should be able to focus on the next input in the form solely through the use of the keyboard, with tab being the expected way to achieve this.

### Reproducing on login page
1. Visit the Login Page
2. Press "TAB" to focus on the password input
3. Press "TAB" again to try focus on the "Show password" button
4. Press "TAB" again to try focus

### Reproducing using entity picker
1. Open a dialog which contains the entity picker
2. Press"TAB" to go to the dropdown button (or clear button if there is already an entity)
3. Press"TAB" again to go to the next field.

## Environment
- Home Assistant release with the issue: 0.116.0.dev0
- Last working Home Assistant release (if known): n/a
- UI Type (States or Lovelace): n/a
- Browser and browser version: Chrome (85/ Canary 87) & FireFox (81)
- Operating system: macOS

## Additional information
The examples I've tested which display this behaviour are the [password input in `ha-form-string`](https://github.com/JeffersonBledsoe/frontend/blob/f8c5eeab5db76f2a14b6c8b51ddb045049d53599/src/components/ha-form/ha-form-string.ts#L43-L60) and the [`ha-entity-picker`](https://github.com/JeffersonBledsoe/frontend/blob/a2ec878ef0b02076004cac7a22cc3585f287cfac/src/components/entity/ha-entity-picker.ts#L196-L236).

**Comment by JeffersonBledsoe** [comment]
My initial thoughts on this, is that because the icon button is contained within the paper-input, the paper-input is re-setting it's focused value to true when the icon button is focused, causing this loop of never being able to move past the icon button.
